core/src/test/java/org/apache/calcite/util/SourceTest.java

```diff
@@ -67,12 +70,27 @@ private static String getRootPrefix() {
     }
   }
 
+  @Test public void testFileToUrl() throws URISyntaxException {
+    final String path = "abc.txt";
+    Source fileSource = file(null, path);
+    URL url = fileSource.url();
+
+    assertFalse(url == null);
+
+    String urlPath = new File(url.toURI().getPath()).getAbsolutePath();
+    assertEquals(absolutePath(path), urlPath);
```

Comment:
Please add the relevant message to clarify why you expect one thing to be equal to another thing.

Reply:
Message added. Please take a look. Thank you.